src/engine/source/builder/src/builders/opfilter/opBuilderHelperFilter.cpp

```diff
@@ -1601,4 +1602,108 @@ FilterOp opBuilderHelperIsTestSession(const Reference& targetField,
     };
 }
 
+// <field>: +keys_exist_in_list/$<list_value>|$<list_reference>
+FilterOp opBuilderHelperKeysExistInList(const Reference& targetField,
```

Comment:
Instead of copying the target field object and iterating over it, we should iterate over the parameter array and check that target field has all keys. Also we can preemptively check sizes to quickly exit.

Reply:
I think that checking the size to exit earlier is fine. Regarding your other proposal, it does not meet the helepr requirement, all the keys in the target_field must be in the list, but not all the elements in the list must be in the target_field.